Hashes.cpp

```diff
@@ -770,19 +770,38 @@ void clhash_seed_init(size_t seed)
 
       // The output is 64 bits, and we consider the input 64 bit as well,
       // so our intermediate values are 128.
-      __uint128_t h = (__uint128_t)seed ^ multiply_shift_r;
       // We mix in len_bytes in the basis, since smhasher considers two keys
       // of different length to be different, even if all the extra bits are 0.
       // This is needed for the AppendZero test.
-      h ^= (__uint128_t)len_bytes << 64;
+      __uint128_t h = (__uint128_t)(seed + len_bytes) * multiply_shift_r;
       for (int i = 0; i < len; i++)
          h += multiply_shift_random[i & MULTIPLY_SHIFT_RANDOM_WORDS-1] * (__uint128_t)buf64[i];
 
-      // Get the last bytes when things are unaligned
+      // Now get the last bytes when things are unaligned
+
+      // This method is slowest:  34.637 cycles/hash
+      /*
+      if (len_bytes & 7) {
+         uint64_t last;
```

Comment:
last = 0

Reply:
Ah right, it doesn't improve performance of the method though.